### PR TITLE
Updates the style guide to include the proper use of the `::` operator

### PR DESCRIPTION
**Pull request by SuperNovaa41**
As the title says.

Feel free to suggest changes if you feel like my addition is not well detailed / accurate enough.

### CODE REVIEW

**Comment by Mothblocks** [comment]
@tgstation/commit-access for style guide changes

**Review by Mothblocks — APPROVED** [review]

**Comment by Mothblocks** [comment]
i agree

File: .github/guides/STYLE.md [hunk 1]
```diff
@@ -322,6 +322,18 @@ It's used within the definition of a datum as opposed to `Initialize` or other p
...
+```
+
+```dm
+/obj/item/fork
+	flags_1 = parent_type::flags_1 | FLAG_COOLER
+```
+
+
 It's important to note that `::` does not apply to every application of `initial()`.
 Primarily in cases where the type you're using for the initial value is not static.
```
Comment (LemonInTheDark) [inline .github/guides/STYLE.md:338]: is this how we feel about this? I'm unsure myself

Comment (SuperNovaa41) [reply .github/guides/STYLE.md:338]: not sure, as non static use is equivalent to `initial(A:B)` and the `:` operator is explicitly disallowed.. so not sure

Comment (optimumtact) [reply .github/guides/STYLE.md:338]: You're just replicating byond docs at this point? why

Comment (SuperNovaa41) [reply .github/guides/STYLE.md:338]: i figured it was useful to have

**Review by Ghommie — APPROVED** [review]

**Review by MrsTonedOne — APPROVED** [review]

**Pull request merged**